packages/framework/theming-react-native/src/BaselinePlatformDefaults.ts

```diff
@@ -1,46 +1,60 @@
-import { ITheme } from './Theme.types';
-import { getStockWebPalette, ITypography, ISpacing } from '@uifabricshared/theming-ramp';
-
-function _defaultTypography(): ITypography {
-  return {
-    sizes: {
-      xxxSmall: 8,
-      xxSmall: 9,
-      xSmall: 10,
-      small: 11,
-      medium: 12,
-      large: 14,
-      xLarge: 18,
-      xxLarge: 24,
-      xxxLarge: 32
-    },
-    weights: {
-      light: '200',
-      semiLight: '300',
-      medium: '500',
-      semiBold: '700',
-      bold: '900'
-    },
-    families: {
-      primary: 'Segoe UI',
-      secondary: 'System',
-      cursive: 'System',
-      monospace: 'System',
-      sansSerif: 'System',
-      serif: 'System'
-    }
-  };
-}
-
-export function defaultSpacing(): ISpacing {
-  return { s2: '4px', s1: '8px', m: '16px', l1: '20px', l2: '32px' };
-}
-
-export function getBaselinePlatformTheme(): ITheme {
-  return {
-    colors: getStockWebPalette(),
-    typography: _defaultTypography(),
-    spacing: defaultSpacing(),
-    components: {}
-  };
-}
+import { ITheme } from './Theme.types';
+import { getStockWebPalette, ITypography, ISpacing } from '@uifabricshared/theming-ramp';
+
+function _defaultTypography(): ITypography {
+  return {
+    sizes: {
+      xxxSmall: 8,
+      xxSmall: 9,
+      xSmall: 10,
+      small: 11,
+      medium: 12,
+      large: 14,
+      xLarge: 18,
+      xxLarge: 24,
+      xxxLarge: 32
+    },
+    weights: {
+      light: '300',
+      normal: '400',
+      semiBold: '700',
+      bold: '900'
+    },
+    families: {
+      primary: 'Segoe UI',
+      secondary: 'System',
+      cursive: 'System',
+      monospace: 'System',
+      sansSerif: 'System',
+      serif: 'System'
+    },
+    variants: {
+      smallStandard: { face: 'Segoe UI', size: 11, weight: '400' },

```

Comment:
I think we spoke about having these defined as something like:
 smallStandard: { face: 'primary', size: 'small', weight: 'normal' }
That way updates to the base ramps will update the variants as well.

Reply:
I'd like to optimize it that way but I'm unsure how. I can't reference those values in the way you have it written. Not all of those values are strings. For instance, size is expected to be a number and weight is expected to be a string but that of '100' through '900' in increments of 100. How can I reference the values by the way they're defined above?

Reply:
There are a couple of parts of it.  First the values reference the name of the key in the typeramp rather than the values.  Then you export a function that resolves types against a theme that people can use in their style processing.  The function would probably look something like:

```
export function resolveTextTokens(theme: ITheme, tokens: ITextTokens): ITextTokens {
  const typography = theme.typography;
  // grab a variant if set, could provide a default value if we want but that means values always get set
  const variant = tokens.variant && variants[tokens.variant];
  const fontSize = tokens.fontSize || (variant && (typography.sizes[variant.size] || variant.size));
  // repeat for others

  // use spread trick to avoid setting fontSize: undefined, unnecesary if there is always a default variant
  return {
    ...(fontSize && { fontSize }) // may need an || {}, can't remember
    ...(fontFamily && { fontFamily })
    ...(fontWeight && { fontWeight })
 };
}
```

This basically returns a style fragment that can be mixed in for style creation.

Reply:
Thanks for the clarification - I forgot about adding the case where variant needs to be looked up by key. I updated the _buildTextStyles function to include this case. It looks a little different than what you wrote here but I think it achieves the same result. Let me know if I can optimize it or make it a little easier to read. The order of precedence for resolving text tokens is now 1. token key 2. token concrete 3. variant key 4. variant concrete.